ci-operator/step-registry/storage/conf/storageclass/set-default-storageclass/storage-conf-storageclass-set-default-storageclass-commands.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+function set_proxy () {
+    if test -s "${SHARED_DIR}/proxy-conf.sh" ; then
+        echo "setting the proxy"
+        echo "source ${SHARED_DIR}/proxy-conf.sh"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "no proxy setting."
+    fi
+}
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+function get_current_sc() {
+    run_command "oc get sc"
+}
+
+function get_default_sc() {
+    CURRENT_DEFAULT_SC=$(oc get sc -ojsonpath='{.items[?(@.metadata.annotations.storageclass\.kubernetes\.io/is-default-class=="true")].metadata.name}')
+    echo "Current default storageclass is:" "${CURRENT_DEFAULT_SC}"
+}
+
+function remove_all_default_sc_annotations () {
+  if [ -z "${CURRENT_DEFAULT_SC}" ]
+  then
+      echo "Currently no default storageclass is set, no need to remove"
+  else
+      for SC in ${CURRENT_DEFAULT_SC}
+      do
+        oc annotate sc "${SC}" "storageclass.kubernetes.io/is-default-class=false" --overwrite
+      done
+  fi
+}
+
+function set_cluster_csi_driver_sc_state_unmanaged () {
+  local ret=0
+  local provisioner=""
+  run_command "oc get sc ${EXPECTED_DEFAULT_STORAGECLASS}" || ret=$?
+  if [[ ! $ret -eq 0 ]]; then
+      echo "Storageclass ${EXPECTED_DEFAULT_STORAGECLASS} is not exist, skip set its clustercsidriver 'storageClassState' as 'Unmanaged' state."
+  else
+      provisioner=$(oc get sc/"${EXPECTED_DEFAULT_STORAGECLASS}" -ojsonpath='{.provisioner}')
+      run_command "oc get clustercsidriver ${provisioner}" || ret=$?
+      if [[ ! $ret -eq 0 ]]; then
+        echo "clustercsidriver/${provisioner} is not exist, skip set 'storageClassState' as 'Unmanaged' state."
+      else
+          oc patch clustercsidriver "${provisioner}" -p '[{"op":"replace","path":"/spec/storageClassState","value":"Unmanaged"}]'  --type json
+      fi  
+  fi
+}
+
+function set_expected_sc_as_default () {
+  ret=0
+  run_command "oc get sc ${EXPECTED_DEFAULT_STORAGECLASS}" || ret=$?
+  if [[ ! $ret -eq 0 ]]; then
+      echo "Storageclass ${EXPECTED_DEFAULT_STORAGECLASS} is not exist, skip setting."
+  else
+      oc annotate sc "${EXPECTED_DEFAULT_STORAGECLASS}" "storageclass.kubernetes.io/is-default-class=true" --overwrite
+  fi
+}
+
+set_proxy
+get_default_sc
+set_cluster_csi_driver_sc_state_unmanaged
+remove_all_default_sc_annotations
```

Comment:
Why must be the operators marked as `Unmanaged`? They will not overwrite the value of `storageclass.kubernetes.io/is-default-class` annotation, they only add it back when it's missing. 

The whole script could be:
```
oc annotate storageclass.kubernetes.io/is-default-class=false --all --overwrite
oc annotate ${EXPECTED_DEFAULT_STORAGECLASS} storageclass.kubernetes.io/is-default-class=true --overwrite
```

Reply:
Hmm, we have some auto test cases will delete the managed storageClasses, so after it execute maybe role back to the origin status. TBH, currently we don't have a better solution for this. I marked it as `Unmanaged ` so that we could add logic make the disruptive case skip executed. I think I could  also just add a label to the csidriver mark it as CI-preset to process the skip logic.

Reply:
This script should just change the default SC, at least that's what the description says. If you want to do other things, like deleting the managed SC, you should create a different step that does just that.

Reply:
Agree, thanks for the suggestion!

Reply:
Fixed.